include/assertion.h

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) 2022 Felix Gohla, Konrad Hanff, Tobias Kantusch,
+ *                    Quentin Kuth, Felix Roth. All rights reserved.
+ *
+ * Use of this source code is governed by a BSD-style
+ * license that can be found in the LICENSE file.
+ */
+
+#pragma once
+
+#include "dev.h"
+#include "largeblob.h"
+// TODO: Use Monocypher with abstraction
+//#include "sha256.h"
+
+// ed25519 signatures are 512 bits long
+// We do not support other (longer) signatures for now.
+#define ASSERTION_SIGNATURE_LENGTH 64
+
+// The standard says 1023, see https://github.com/w3c/webauthn/pull/1664.
+// see https://github.com/solokeys/fido-authenticator/pull/8
+// We define this as our maximum in order to reduce stack usage.
+#define ASSERTION_MAX_KEY_HANDLE_LENGTH 255
+
+#define SHA256_DIGEST_SIZE 32
+// This is not defined by the standard.
+// However, we define this as our limit.
+#define ASSERTION_AUTH_DATA_LENGTH 128
+#define ASSERTION_AUTH_DATA_RPID_HASH_LEN SHA256_DIGEST_SIZE
+
+// A SHA256 hash.
+#define ASSERTION_CLIENT_DATA_HASH_LEN SHA256_DIGEST_SIZE
+
+#define ASSERTION_PRE_IMAGE_LENGTH ASSERTION_AUTH_DATA_LENGTH + SHA256_DIGEST_SIZE
+
+typedef void es256_pk_t;
+
+typedef struct fido_assert_blob {
+    uint8_t const  *ptr;
+    size_t          len;
+} fido_assert_blob_t;
```

Comment:
It has this extra attribute of max length, doesn't it?

Reply:
Yes. max length would not be needed. Maybe the name `assert_blob` seems strange to me since it is a pointer to a buffer with associated buffer length. This has nothing to do with the assertion (or FIDO). We could name it `fido_buffer` or something else.